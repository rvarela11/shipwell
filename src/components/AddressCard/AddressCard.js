// @vendors
import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';

// @material-ui
import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

// @components
import FromAddressInput from '../../containers/FromAddressInput';
import ToAddressInput from '../../containers/ToAddressInput';

// Setting material-ui classes
const styles = theme => ({
    address_card: {
        width: '60%',
        height: '50vh',
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        [theme.breakpoints.between('xs', 'sm')]: {
            width: '75%'
        }
    },
    address_card_content: {
        width: '75%'
    },
    address_title: {
        textAlign: 'center'
    },
    typography_media_query: {
        [theme.breakpoints.between('xs', 'sm')]: {
            fontSize: '1.25rem'
        }
    }
});

const AddressCard = (props) => {
    const { classes } = props;
    return (
        <Card className={classes.address_card}>
            <CardContent className={classes.address_card_content}>
                <Typography className={`${classes.typography_media_query} ${classes.address_title}`} variant="h5">
                    Address
                </Typography>
                <FromAddressInput />
                <ToAddressInput />
                <CardActions>
                    <Button
                        color="primary"
                        size="large"
                    >
                        Search
                    </Button>
                </CardActions>
            </CardContent>
        </Card>
    );
};

AddressCard.propTypes = {
    classes: PropTypes.object.isRequired
};

export default withStyles(styles)(AddressCard);
