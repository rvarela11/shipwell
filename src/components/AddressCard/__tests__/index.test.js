// @vendors
import React from 'react';
import { shallow } from 'enzyme';

// @components
import AddressCard from '../AddressCard';

describe('AddressCard', () => {
    it('renders AddressCard without crashing', () => {
        shallow(<AddressCard />);
    });
});
