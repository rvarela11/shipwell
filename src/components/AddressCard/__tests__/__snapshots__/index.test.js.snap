// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressCard renders AddressCard without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddressCard
    classes={
      Object {
        "address_card": "AddressCard-address_card-1",
        "address_card_content": "AddressCard-address_card_content-2",
        "address_title": "AddressCard-address_title-3",
        "typography_media_query": "AddressCard-typography_media_query-4",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(CardContent)
        className="AddressCard-address_card_content-2"
      >
        <WithStyles(Typography)
          className="AddressCard-typography_media_query-4 AddressCard-address_title-3"
          variant="h5"
        >
          Address
        </WithStyles(Typography)>
        <Connect(AddressInput) />
        <Connect(AddressInput) />
        <WithStyles(CardActions)>
          <Connect(WithStyles(CardButton)) />
        </WithStyles(CardActions)>
      </WithStyles(CardContent)>,
      "className": "AddressCard-address_card-1",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(Typography)
            className="AddressCard-typography_media_query-4 AddressCard-address_title-3"
            variant="h5"
          >
            Address
          </WithStyles(Typography)>,
          <Connect(AddressInput) />,
          <Connect(AddressInput) />,
          <WithStyles(CardActions)>
            <Connect(WithStyles(CardButton)) />
          </WithStyles(CardActions)>,
        ],
        "className": "AddressCard-address_card_content-2",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Address",
            "className": "AddressCard-typography_media_query-4 AddressCard-address_title-3",
            "variant": "h5",
          },
          "ref": null,
          "rendered": "Address",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(WithStyles(CardButton)) />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(CardContent)
          className="AddressCard-address_card_content-2"
        >
          <WithStyles(Typography)
            className="AddressCard-typography_media_query-4 AddressCard-address_title-3"
            variant="h5"
          >
            Address
          </WithStyles(Typography)>
          <Connect(AddressInput) />
          <Connect(AddressInput) />
          <WithStyles(CardActions)>
            <Connect(WithStyles(CardButton)) />
          </WithStyles(CardActions)>
        </WithStyles(CardContent)>,
        "className": "AddressCard-address_card-1",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Typography)
              className="AddressCard-typography_media_query-4 AddressCard-address_title-3"
              variant="h5"
            >
              Address
            </WithStyles(Typography)>,
            <Connect(AddressInput) />,
            <Connect(AddressInput) />,
            <WithStyles(CardActions)>
              <Connect(WithStyles(CardButton)) />
            </WithStyles(CardActions)>,
          ],
          "className": "AddressCard-address_card_content-2",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Address",
              "className": "AddressCard-typography_media_query-4 AddressCard-address_title-3",
              "variant": "h5",
            },
            "ref": null,
            "rendered": "Address",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Connect(WithStyles(CardButton)) />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
